### PR TITLE
fix: respect isCustomElement

### PR DESCRIPTION
**Pull request by fxy060608**
### 🤔 What is the nature of this change?

- [ ] New feature
- [x] Fix bug
- [ ] Style optimization
- [ ] Code style optimization
- [ ] Performance optimization
- [ ] Build optimization
- [ ] Refactor code or style
- [ ] Test related
- [ ] Other

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

File: packages/babel-plugin-jsx/src/utils.ts [hunk 1]
```diff
@@ -50,6 +50,10 @@ export const checkIsComponent = (path: NodePath<t.JSXOpeningElement>): boolean =
 
   const tag = (namePath as NodePath<t.JSXIdentifier>).node.name;
 
+  if (state.opts.isCustomElement && state.opts.isCustomElement(tag)) {
+    return false;
+  }
+
   return shouldTransformedToSlots(tag) && !htmlTags.includes(tag) && !svgTags.includes(tag);
```
Comment (Amour1688) [inline packages/babel-plugin-jsx/src/utils.ts:57]:
```suggestion
  return  state.opts.isCustomElement?.(tag) && shouldTransformedToSlots(tag) && !htmlTags.includes(tag) && !svgTags.includes(tag);
```

Comment (fxy060608) [reply packages/babel-plugin-jsx/src/utils.ts:57]: state.opts.isCustomElement?.(tag) => Type 'boolean | undefined' is not assignable to type 'boolean'

Comment (Amour1688) [reply packages/babel-plugin-jsx/src/utils.ts:57]: sorry
You can try `!!state.opts.isCustomElement?.(tag)`.

Comment (fxy060608) [reply packages/babel-plugin-jsx/src/utils.ts:57]: !!!(state.opts.isCustomElement?.(tag)) => error  Redundant double negation  no-extra-boolean-cast
add /* eslint-disable no-extra-boolean-cast */ ?

Comment (Amour1688) [reply packages/babel-plugin-jsx/src/utils.ts:57]: `!!` not `!!!`

Comment (fxy060608) [reply packages/babel-plugin-jsx/src/utils.ts:57]: checkIsComponent should be false when isCustomElement is true

**Pull request merged**